datafusion/expr/src/operator.rs

```diff
@@ -79,6 +79,26 @@ pub enum Operator {
     StringConcat,
 }
 
+impl Operator {
+    /// If the operator can be negated, return the negated operator
+    /// otherwise return None
+    pub fn negate(&self) -> Option<Operator> {
+        match self {
+            Operator::Eq => Some(Operator::NotEq),
+            Operator::NotEq => Some(Operator::Eq),
+            Operator::Lt => Some(Operator::GtEq),
+            Operator::LtEq => Some(Operator::Gt),
+            Operator::Gt => Some(Operator::LtEq),
+            Operator::GtEq => Some(Operator::Lt),
+            Operator::Like => Some(Operator::NotLike),
+            Operator::NotLike => Some(Operator::Like),
+            Operator::IsDistinctFrom => Some(Operator::IsNotDistinctFrom),
+            Operator::IsNotDistinctFrom => Some(Operator::IsDistinctFrom),
+            _ => None,
```

Comment:
I would like to suggest replacing `_` with an an explicit enumeration of all the other operators, like

```rust
Operator::LeftShift | OperatiorRightShift => None
```

The reason is so that as new operators are added, the compiler will point us at this code and we will get another chance to avoid bugs like https://github.com/apache/arrow-datafusion/pull/3167/files#r946145318